RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractEdge.java

```diff
@@ -0,0 +1,197 @@
+/*
+ *
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.recentactivity;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import org.openide.modules.InstalledFileLocator;
+import org.openide.util.NbBundle;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.coreutils.ExecUtil;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.coreutils.PlatformUtil;
+import org.sleuthkit.autopsy.datamodel.ContentUtils;
+import org.sleuthkit.autopsy.ingest.DataSourceIngestModuleProcessTerminator;
+import org.sleuthkit.autopsy.ingest.IngestJobContext;
+import org.sleuthkit.autopsy.ingest.IngestServices;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+public class ExtractEdge extends Extract{
+    
+    private static final Logger logger = Logger.getLogger(ExtractIE.class.getName());
+    private final IngestServices services = IngestServices.getInstance();
+    private final String moduleTempResultsDir;
+    private Content dataSource;
+    private IngestJobContext context;
+    
+    private static String ESE_TOOL_NAME = "ESEDatabaseView.exe";
+    private static File ESE_TOOL_FILE;
+    private static String EDGE_WEBCACHE_NAME = "WebCacheV01.dat";
+    private static String EDGE = "Edge";
+
+    ExtractEdge() throws NoCurrentCaseException{
+        moduleName = NbBundle.getMessage(Chrome.class, "ExtractEdge.moduleName");
+        moduleTempResultsDir = RAImageIngestModule.getRATempPath(Case.getCurrentCaseThrows(), EDGE) + File.separator + "results";
+    }
+    
+    @Override
+    void process(Content dataSource, IngestJobContext context) {
+        this.dataSource = dataSource;
+        this.context = context;
+        dataFound = false;
```

Comment:
I see that this dataFound field is yet another protected field in the super class. For this story, let's do the partial fix of adding a protected setter for the super class field and then this new code can use that instead of writing directly to the field.